### PR TITLE
Implement ReferenceCell::vertex_faces()

### PR DESCRIPTION
**Pull request by peterrum**
needed by #11575

### CODE REVIEW

File: include/deal.II/grid/reference_cell.h [hunk 1]
```diff
@@ -182,6 +189,78 @@ namespace ReferenceCell
...
 
 
+  inline ArrayView<const unsigned int>
+  Type::vertex_faces(unsigned int vertex) const
+  {
+    if (kind == Type::Line)
+      {
+        AssertIndexRange(vertex, GeometryInfo<1>::vertices_per_cell);
+        return ArrayView<const unsigned int>(
+          &GeometryInfo<2>::vertex_to_face[vertex][0], 1);
```
Comment (kronbichler) [inline include/deal.II/grid/reference_cell.h:199]: Can you take a look at the tidy error
```
error: avoid repeating the return type from the declaration; use a braced initializer list instead [modernize-return-braced-init-list,-warnings-as-errors]
```

File: include/deal.II/grid/reference_cell.h [hunk 2]
```diff
@@ -108,6 +109,12 @@ namespace ReferenceCell
     void
     serialize(Archive &archive, const unsigned int /*version*/);
 
+    /**
+     * Return a vector of faces a @p vertex belongs to.
+     */
+    ArrayView<const unsigned int>
+    vertex_faces(unsigned int vertex) const;
```
Comment (bangerth) [inline include/deal.II/grid/reference_cell.h:116]:
```suggestion
    vertex_faces(const unsigned int vertex) const;
```
Same in the definition.

Comment (bangerth) [reply include/deal.II/grid/reference_cell.h:116]: Separately, I'm not in love with the name of the function. Maybe `faces_for_given_vertex()`?

**Review by bangerth — COMMENTED** [review]
Thanks for already putting the function inside `ReferenceCell::Type` :-)

**Comment by peterrum** [comment]
> Thanks for already putting the function inside ReferenceCell::Type :-)

**Review by bangerth — APPROVED** [review]

**Review by kronbichler — APPROVED** [review]

**Pull request merged**